sdk/src/main/java/io/radar/sdk/util/BatteryState.kt

```diff
@@ -0,0 +1,113 @@
+package io.radar.sdk.util
+
+import android.os.PowerManager
+import io.radar.sdk.RadarBatteryManager
+
+/**
+ * Contains information about the battery state
+ */
+internal data class BatteryState(
+    val isCharging: Boolean,
+    val percent: Float,
+    val powerSaveMode: Boolean?,
+    val isIgnoringBatteryOptimizations: Boolean,
+    val locationPowerSaveMode: Int,
+    val isDeviceIdleMode: Boolean
+) {
+
+    /**
+     * The performance state. This uses Android's [PowerManager] to gather information
+     * about the current optimization settings that could affect the SDK's performance, and returns
+     * a simplified status which identifies whether or not battery optimizations may be affecting
+     * the SDK's responsiveness.
+     */
+    val performanceState: PerformanceState
+
+    init {
```

Comment:
Are there any links to Android docs we can add to the comment that explains this logic?

Reply:
No - this is original.